### PR TITLE
Remove unneeded type: ignore comments

### PR DESCRIPTION
**Pull request by kclowes**
### What was wrong?
We had a bunch of lint errors that suddenly popped up. 

### How was it fixed?
Removed unneeded `# type: ignore` comments

### Todo:
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md)

#### Cute Animal Picture

image

### CODE REVIEW

File: web3/providers/ipc.py [hunk 1]
```diff
@@ -125,8 +125,8 @@ def get_default_ipc_path() -> str:  # type: ignore
 
         base_trinity_path = Path('~').expanduser() / '.local' / 'share' / 'trinity'
         # type ignored b/c ipc_path is already defined as a str above
-        ipc_path = base_trinity_path / 'mainnet' / 'jsonrpc.ipc'  # type: ignore
-        if ipc_path.exists():  # type: ignore
+        ipc_path = base_trinity_path / 'mainnet' / 'jsonrpc.ipc'
+        if ipc_path.exists():
```
Comment (njgheorghita) [inline web3/providers/ipc.py]: I think just try leaving these two `# type: ignore`'s in. These two in particular have always been flaky for me.

Comment (kclowes) [reply web3/providers/ipc.py]: Ah, thanks!

File: web3/_utils/events.py [hunk 2]
```diff
@@ -429,7 +429,7 @@ def _build_argument_filters_from_event_abi(
 
 
 # type ignored b/c mypy complains about curried fns
-array_to_tuple = apply_formatter_if(is_list_like, tuple)  # type: ignore
+array_to_tuple = apply_formatter_if(is_list_like, tuple)
```
Comment (njgheorghita) [inline web3/_utils/events.py]: For pretty much all of these, we can also delete the comments directly above

**Comment by njgheorghita** [comment]
@kclowes Looks like this pr is good to close now. I guess what happened is I had these updates in #1538 with the `eth-utils` dependency update, but the web3 master build picked up the newest available `eth-utils` version before those changes got merged in. But they're merged in now, and we have a green circle badge again! 🎆

**Comment by kclowes** [comment]
weeeeird. But mystery solved! Thanks @njgheorghita!

**Pull request closed without merging**